File: apps/mobile/app/src/main/java/com/basehaptic/mobile/data/model/Game.kt
package com.basehaptic.mobile.data.model

data class Game(
    val id: String,
    val homeTeam: String,
    val awayTeam: String,
    val homeTeamId: Team,
    val awayTeamId: Team,
    val homeScore: Int,
    val awayScore: Int,
    val inning: String,
    val status: GameStatus,
    val time: String? = null,
    val isMyTeam: Boolean = false,
    val weather: GameWeatherSummary? = null,
    val homePitcher: Pitcher? = null,
    val awayPitcher: Pitcher? = null
)

data class GameWeatherSummary(
    val stadiumCode: String,
    val stadiumName: String,
    val stadiumShortName: String,
    val forecastDate: String? = null,
    val forecastTime: String? = null,
    val forecastTimeLabel: String? = null,
    val condition: String,
    val temperatureC: Int? = null,
    val precipitationProbability: Int? = null,
    val precipitationType: String? = null,
    val windSpeedMps: Double? = null,
    val isIndoor: Boolean = false,
    val displayText: String
)

data class Pitcher(
    val name: String,
    val winStreak: Int,
    val record: PitcherRecord
)

data class PitcherRecord(
    val wins: Int,
    val draws: Int,
    val losses: Int
)

enum class GameStatus {
    LIVE,
    SCHEDULED,
    FINISHED,
    CANCELED,
    POSTPONED
}

File: ios/mobile/BaseHaptic/Models/Game.swift
import Foundation

enum GameStatus: String, Codable {
    case live = "LIVE"
    case scheduled = "SCHEDULED"
    case finished = "FINISHED"
    case canceled = "CANCELED"
    case postponed = "POSTPONED"
}

struct Game: Identifiable, Codable {
    let id: String
    let homeTeam: String
    let awayTeam: String
    let homeTeamId: Team
    let awayTeamId: Team
    let homeScore: Int
    let awayScore: Int
    let inning: String
    let status: GameStatus
    let time: String?
    let isMyTeam: Bool
    let weather: GameWeatherSummary?
    let homePitcher: Pitcher?
    let awayPitcher: Pitcher?

    init(
        id: String,
        homeTeam: String,
        awayTeam: String,
        homeTeamId: Team,
        awayTeamId: Team,
        homeScore: Int,
        awayScore: Int,
        inning: String,
        status: GameStatus,
        time: String? = nil,
        isMyTeam: Bool = false,
        weather: GameWeatherSummary? = nil,
        homePitcher: Pitcher? = nil,
        awayPitcher: Pitcher? = nil
    ) {
        self.id = id
        self.homeTeam = homeTeam
        self.awayTeam = awayTeam
        self.homeTeamId = homeTeamId
        self.awayTeamId = awayTeamId
        self.homeScore = homeScore
        self.awayScore = awayScore
        self.inning = inning
        self.status = status
        self.time = time
        self.isMyTeam = isMyTeam
        self.weather = weather
        self.homePitcher = homePitcher
        self.awayPitcher = awayPitcher
    }
}

struct GameWeatherSummary: Codable {
    let stadiumCode: String
    let stadiumName: String
    let stadiumShortName: String
    let forecastDate: String?
    let forecastTime: String?
    let forecastTimeLabel: String?
    let condition: String
    let temperatureC: Int?
    let precipitationProbability: Int?
    let precipitationType: String?
    let windSpeedMps: Double?
    let isIndoor: Bool
    let displayText: String
}

struct Pitcher: Codable {
    let name: String
    let winStreak: Int
    let record: PitcherRecord
}

struct PitcherRecord: Codable {
    let wins: Int
    let draws: Int
    let losses: Int
}
